proto/api/v1/audit.proto

```diff
@@ -0,0 +1,110 @@
+import "google/protobuf/timestamp.proto";
+
+// AuditService serves audit related functions
+//
+service AuditService {
+  // Get a audit
+  rpc Get(AuditServiceGetRequest) returns (AuditServiceGetResponse) {
+    option (tenant_roles) = TENANT_ROLE_OWNER;
+    option (auditing) = AUDITING_EXCLUDED;
+  }
+  // List your audit traces
+  rpc List(AuditServiceListRequest) returns (AuditServiceListResponse) {
+    option (tenant_roles) = TENANT_ROLE_OWNER;
+    option (auditing) = AUDITING_EXCLUDED;
+  }
+}
+
+// AuditTrace is an audit trace
+message AuditTrace {
+  // Uuid of the audit trace
+  string uuid = 1 [(buf.validate.field).string.uuid = true];
+  // Timestamp is the timestamp when the request arrived at the api
+  google.protobuf.Timestamp timestamp = 2;
+  // User is the user who called the api method
+  string user = 3;
+  // Tenant is the tenant targeted by the api call
+  string tenant = 4;
+  // Project is the project targeted by the api call
+  optional string project = 5;
+  // Method is the api method that was called
+  string method = 6;
+  // Request is the payload of the request
+  optional string body = 7;
+  // Response is the payload of the response
+  string source_ip = 8;
+  // ResultCode is a string describing the result of the api call
+  optional int32 result_code = 9;
+  // Phase is a string representing the request phase
+  string phase = 10;
+}
+
+// AuditServiceListRequest is the request payload to list audits
+message AuditServiceListRequest {
+  // Login of the tenant
+  string login = 1 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Uuid of the audit trace
+  optional string uuid = 2 [(buf.validate.field).string.uuid = true];
+  // From describes the start of the time window in which to list audit traces
+  google.protobuf.Timestamp from = 3 [(buf.validate.field).required = true];
+  // To describes the end of the time window in which to list audit traces
+  google.protobuf.Timestamp to = 4 [(buf.validate.field).required = true];
+  // User is the user who called the api method
+  optional string user = 5 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Project is the project targeted by the api call
+  optional string project = 6 [(buf.validate.field).string.uuid = true];
+  // Method is the api method that was called
+  optional string method = 7 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 256
+  }];
+  // SourceIP contains the ip address of the caller
+  optional string source_ip = 8 [(buf.validate.field).string.ip = true];
+  // ResultCode is a string describing the result of the api call
+  optional int32 result_code = 9 [(buf.validate.field).int32 = {gte: 0}];
+  // Body is a string providing text-search of the body field
+  optional string body = 10 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 256
+  }];
+  // Limit is a number limiting the length of the response (min: 1, max: 1000)
```

Comment:
Should we default this? Otherwise the responses may be too large (user that request all audit traces).